src/Blob/Models/BlobProperties.php

```diff
@@ -24,8 +24,13 @@ public function __construct(
 
     public static function fromResponseHeaders(ResponseInterface $response): self
     {
+        $lastModified = \DateTimeImmutable::createFromFormat(\DateTimeInterface::RFC1123, $response->getHeaderLine('Last-Modified'));
+        if ($lastModified === false) {
+            throw new \Exception("Should not happen!");
```

Comment:
Lets bring a little better description here since when it does happen an user has no clue where to look

Reply:
will fix in next pr